### DISCUSSION TITLE
How to get all profile user info from Okta

### DISCUSSION THREAD

**Discussion by AutoMates-Joep** [nextauthjs/next-auth#8201] [Help]
### Question 💬

I implemented AuthJS for Sveltekit together with Okta as provider.
I can login and logout without any problems.

After logging in I get the username and email in the session info. How can I get all other Okta profile fields?

### How to reproduce ☕️
**This is my hooks.server.js file:**
`export const handle = sequence(
	SvelteKitAuth({
		providers: [Okta({
      clientId: OKTA_CLIENT_ID,
      clientSecret: OKTA_CLIENT_SECRET,
      issuer: OKTA_ISSUER,
      authorization: {
        params: {
          state: 'login',
        }
      }
    })],
    secret: AUTH_SECRET,
    debug: true,
	}),
	authorization
);`

**In +layout.server.js I can log the session:**
`console.log(await locals.getSession())`

**And this are the only user profile fields I receive:**
`{
  user: { name: 'Username', email: 'user@test.com },
  expires: '2023-08-01T19:40:03.075Z'
}`

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by ThangHuuVu** [comment]
I think you can follow the following steps:
1. Use the [account callback](https://authjs.dev/reference/core/providers_oauth#account) in the provider options to customize the params return from the OAuth's userinfo endpoint.
2. In the [JWT callback](https://authjs.dev/reference/core/types#callbacksoptions), get the data from the `user` object and pass it down to the `session` callback afterward by assigning to `token`


Code will look like this:
```ts
export const handle = SvelteKitAuth({
  providers: [Okta({ clientId: OKTA_ID, clientSecret: OKTA_SECRET, issuer: OKTA_ISSUER, account: (tokens) => token })],
  callbacks: {
    async jwt(token, user, account, profile, isNewUser) {
      // assign user to token
      if (user) {
        token.user = user
      }
      return token
    },
    async session(session, token) {
      // assign user to session
      session.user = token.user
      return session
    }
  }
})
```

let me know if this helps 🙏

**Comment by HChenCBRE** [comment]
You can add in signIn function inside callbacks

signIn: ({
            user,
            account,
            profile,
            email
        } : any) => {
            // this include both id_token and access_token
            console.log(account)
            //this include the decoded id_token, which includes the profile data
            console.log(profile)
            return true;
        }
